### PR TITLE
 Add To Cart with Options: introduce blockified feature flag

### LINKED ISSUE #53007: [Enhancement]: Add Feature Flag for Blockified Add to Cart Form
### Description

Introduce a feature flag to `enable` or `disable` the blockified Add to Cart Form. This flag will allow developers and site administrators to toggle the new blockified experience without affecting the existing functionality.

### Details

**Feature Flag Name**: `blockified-add-to-cart-form`

```[tasklist]
### Acceptance Criteria
- [ ] The `Add to Cart Form` conditionally renders based on the feature flag status.
```

### PR DESCRIPTION
**Pull request by retrofox**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR introduces the `blockify-add-to-cart` feature flag.
Additionally, it adds a new panel to the block sidebar to show all enabled features.
Let's keep in mind that the stepper mode is also visible depending on a feature flag.

Closes #53007

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Ensure install and activate the `WooCommerce Beta Tester` plugin
image

2. Go to the `WooCommerce Admin Test Helper` -> `Features` page

image

3. Confirm the `blockify-add-to-cart` is listed there. Also, identify the `add-to-cart-with-options-stepper-layout` feature.
Screenshot 2024-11-26 at 7 17 58 AM

4. Enable `blockify-add-to-cart`
5. Edit the `Product individual` template
image
6. Select the `Add to Cart with Options` block
7. Confirm, that when one of the feature flags is enabled, the sidebar shows the Development panel

image

8. Confirm when no one of these feature flags is enabled, the Dev panel isn't shown.
image

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/assets/js/blocks/product-elements/add-to-cart-form/edit.tsx [hunk 1]
```diff
@@ -17,15 +17,32 @@ import { isBoolean } from '@woocommerce/types';
...
+};
+
+export type UpdateFeaturesType = ( key: FeaturesKeys, value: boolean ) => void;
+
+// Pick the value of the "blockify add to cart flag"
+const isBlockifyAddToCartEnabled = getSettingWithCoercion(
+	'isBlockifyAddToCartEnabled',
+	false,
+	isBoolean
+);
```
Comment (Aljullu) [inline plugins/woocommerce-blocks/assets/js/blocks/product-elements/add-to-cart-form/edit.tsx]: I guess this is temporary, but wondering if we could define `isStepperLayoutFeatureEnabled` and `isBlockifyAddToCartEnabled` next to each other?

File: plugins/woocommerce-blocks/assets/js/blocks/product-elements/add-to-cart-form/settings.tsx [hunk 2]
```diff
@@ -42,36 +52,66 @@ const getHelpText = ( quantitySelectorStyle: QuantitySelectorStyle ) => {
 	}
 };
 
-export const Settings = ( {
+export const AddToCartSettings = ( {
```
Comment (Aljullu) [inline plugins/woocommerce-blocks/assets/js/blocks/product-elements/add-to-cart-form/settings.tsx]: I would name this component `AddToCartFormSettings`, to keep it consistent with `AddToCartFormEdit`. What do you think?

**Review by Aljullu — COMMENTED** [review]
Working as described!

Personally, I would name the feature flag '`blockified-add-to-cart`'. '`blockify-add-to-cart`' sounds like an action to me, so it might be confusing.

Besides that, I left a couple of minor inline comments.

File: plugins/woocommerce-blocks/assets/js/blocks/product-elements/add-to-cart-form/edit.tsx (hunk 1 above)
Comment (retrofox) [reply plugins/woocommerce-blocks/assets/js/blocks/product-elements/add-to-cart-form/edit.tsx]: yes, we should indeed. The thing is, I didn't want to change something that seems to be out of the scope of this PR. Maybe a follow-up?

**Comment by retrofox** [comment]
> Working as described!

> Personally, I would name the feature flag '`blockified-add-to-cart`'. '`blockify-add-to-cart`' sounds like an action to me, so it might be confusing.

addressed here: 59e00056ed7188b13bdd344da5df222f55342a71

Thanks for the review 🙇‍♂️

**Review by Aljullu — APPROVED** [review]
Thanks for the updates, @retrofox! LGTM.

**Pull request merged**